crates/autopilot/src/run_loop.rs

```diff
@@ -181,15 +221,16 @@ impl RunLoop {
                 competition_simulation_block,
                 call_data,
                 uninternalized_call_data,
+                competition_table,
             };
             tracing::info!(?competition, "saving competition");
             if let Err(err) = self.save_competition(&competition).await {
                 tracing::error!(?err, "failed to save competition");
                 return;
             }
 
-            tracing::info!(url = %self.drivers[index].url, "settling with solver");
-            match self.settle(id, &self.drivers[index], &solution).await {
+            tracing::info!(url = %self.drivers[*index].url, "settling with solver");
+            match self.settle(id, &self.drivers[*index], solution).await {
```

Comment:
This looks like a (pre-existing) bug. Since the indices refer to solutions sorted by score and not by their driver I believe this does will usually tell the wrong driver to execute its solution.

Reply:
AFAIS indexes are paired with solutions making a tuple `(index, solution)` therefore indexes are following it's original solution even after sorting by score.

Drivers: [0,1,2]
Solutions: [(0,A),(1,B),(2,C)]
Sorted solutions example: [(1,B), (2,C), (0,A)]